platforms/core-runtime/internal-instrumentation-processor/src/main/java/org/gradle/internal/instrumentation/extensions/property/PropertyUpgradeAnnotatedMethodReader.java

```diff
@@ -440,7 +467,7 @@ private List<RequestExtra> getJvmRequestExtras(AccessorSpec accessor, Executable
         extras.add(new RequestExtra.OriginatingElement(method));
         extras.add(new RequestExtra.InterceptJvmCalls(interceptorsClassName, BYTECODE_UPGRADE));
         String implementationClass = accessor.generatedClassName;
-        Type newReturnType = TypeUtils.extractReturnType(method);
+        TypeName newPropertyType = TypeName.get(method.getReturnType());
```

Comment:
💭 We now pass `TypeName` instead of `Type` to  `PropertyUpgradeClassSourceGenerator.java`, since it contains also type parameters.

Imo we should everywhere get rid of `Type` and use `TypeName`, but that is a bigger change.

Reply:
That sounds like a good idea. Let's create an issue for that.